libutils/src/debug.c

```diff
@@ -83,8 +83,8 @@ utils_memory_dump(void* address, size_t bytes, int word_size)
             bytes += extra_bytes;
         }
         /* Print each line */
-        for (a = address; a < address + bytes; a += MD_BYTES_PER_LINE) {
-            md_print_line(a, word_size);
+        for (a = (uintptr_t)address; a < (uintptr_t)address + bytes; a += MD_BYTES_PER_LINE) {
```

Comment:
remove `a` and the pointer arithmetic and use:
```
for (size_t i = 0; i < bytes; i += MD_BYTES_PER_LINE) {
    md_print_line( &((uint8_t*)address)[i], word_size);
}
```

Reply:
I updated this